rfcs/0000-single-cluster-dnspolicy.md

```diff
@@ -0,0 +1,286 @@
+
+The `strategy` field is mandatory and dictates what kind of dns record structure the policy will create. Two strategy options are allowed `simple` or `loadbalanced`.
+
+A reconciliation of DNSPolicy processes the target gateway and creates a DNSRecord per listener that is supported by the currently configured provider(hostname matches the hosted zones accessible with the credentials and config). The strategy used will determine the contents of the DNSRecord resources Endpoints array.
+
+#### Simple
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSRecord
+spec:
+  providerRef:
+    name: my-route53-credentials
+    namespace: my-gateways
+    kind: Secret
+  endpoints:
+    - dnsName: myapp.mn.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.31.200.0
+```
+
+#### LoadBalanced
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSRecord
+spec:
+  providerRef:
+    name: my-route53-credentials
+    namespace: my-gateways
+    kind: Secret
+  endpoints:
+    - dnsName: myapp.mn.hcpapps.net
+      recordTTL: 300
+      recordType: CNAME
+      targets:
+        - lb-4ej5le.myapp.mn.hcpapps.net
+    - dnsName: lb-4ej5le.myapp.mn.hcpapps.net
+      providerSpecific:
+        - name: geo-code
+          value: '*'
+      recordTTL: 300
+      recordType: CNAME
+      setIdentifier: default
+      targets:
+        - default.lb-4ej5le.myapp.mn.hcpapps.net
+    - dnsName: default.lb-4ej5le.myapp.mn.hcpapps.net
+      providerSpecific:
+        - name: weight
+          value: "120"
+      recordTTL: 60
+      recordType: CNAME
+      setIdentifier: lrnse3.lb-4ej5le.myapp.mn.hcpapps.net
+      targets:
+        - lrnse3.lb-4ej5le.myapp.mn.hcpapps.net
+    - dnsName: lrnse3.lb-4ej5le.myapp.mn.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.31.200.0
+```
+
+Note: `LoadBalanced` is the current default for DNSPolicy and more details about it can be found in the original [DNSPolicy rfc](https://github.com/Kuadrant/architecture/blob/main/rfcs/0003-dns-policy.md) 
+
+### DNSRecord.spec.providerRef
+
+The DNSRecord API is updated to remove the `managedZone` reference in favour of directly referencing the `providerRef` credentials instead. The DNSRecord reconciliation will be unchanged except for loading the provider client from `providerRef` credentials.
+
+The DNSPolicy reconciliation will be updated to remove the requirement for a ManagedZone resource to be created before a DNSPolicy can create dns records for it, instead it will be replaced in favour of just listing available zones directly in the currently configured dns provider. If no matching zone is found, no DNSRecord will be created.
+
+There is a potential for a DNSRecord to be created successfully, but then a provider updated to remove access. In this case it is the responsibility of the DNSPolicy controller to remove DNSRecords it created that can no longed be reconciled into the configured provider. 
```

Comment:
I wonder would it be better to add something to the status of these rather than remove them?

Reply:
I'm not sure it (the policy controller) should directly update the status of the DNSRecord, the DNSRecord reconciliation will already update that anyway if it can't access the provider for whatever reason. In that scenario it comes back to aggregating the status of the policy created resources into the policy status and/or the gateway status (Separate proposal we talked about).

I agree though it probably shouldn't just remove it though, will update that.